lib/portage/process.py

```diff
@@ -325,6 +326,31 @@ def spawn(
 
     env = os.environ if env is None else env
 
+    env_size = 0
+    env_largest_name = None
+    env_largest_size = 0
+    for env_name, env_value in env.items():
+        env_name_size = len(env_name)
+        env_value_size = len(env_value)
+        total_size = env_name_size + env_value_size
```

Comment:
Shouldn't you be counting `=` and null terminators too?

Reply:
Probably. I wonder if it would even be more accurate to convert the strings to UTF-8 bytes and count the length (if this is the right encoding). Right now, it is probably an approximation, but still sufficient to perform its use case. That said, I am always happy to improve on that. And adding `+ 2` to the `total_size` calculation seems trivial.